tests/integration/test_bundle.py

```diff
@@ -0,0 +1,147 @@
+import asyncio
+import json
+import logging
+
+from pathlib import Path
+
+import aiohttp
+import lightkube
+import lightkube.codecs
+import pytest
+import requests
+import tenacity
+import yaml
+from pytest_operator.plugin import OpsTest
+
+ISVC = lightkube.generic_resource.create_namespaced_resource(
+    group="serving.kserve.io",
+    version="v1beta1",
+    kind="InferenceService",
+    plural="inferenceservices",
+    verbs=None,
+)
+BUNDLE_PATH = "./bundle/bundle.yaml"
+SKLEARN_INF_SVC_YAML = yaml.safe_load(
+    Path("./tests/integration/sklearn-iris.yaml").read_text()
+)
+SKLEARN_INF_SVC_OBJECT = lightkube.codecs.load_all_yaml(
+    yaml.dump(SKLEARN_INF_SVC_YAML)
+)[0]
+SKLEARN_INF_SVC_NAME = SKLEARN_INF_SVC_OBJECT.metadata.name
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="session")
+def lightkube_client() -> lightkube.Client:
+    client = lightkube.Client(field_manager="autoscaling-bundle")
+    return client
+
+
+async def cli_deploy_bundle(ops_test: OpsTest, bundle_path: str):
+    """Deploy bundle from charmhub or from file."""
+    run_args = [
+        "juju",
+        "deploy",
+        "--trust",
+        "-m",
+        ops_test.model_full_name,
+        bundle_path,
+    ]
+
+    retcode, stdout, stderr = await ops_test.run(*run_args)
+    assert retcode == 0, f"Deploy failed: {(stderr or stdout).strip()}"
+    logger.info(stdout)
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    """Deploy the autoscaling-model-server bundle."""
+    await cli_deploy_bundle(ops_test, bundle_path=BUNDLE_PATH)
+    # Configure knative-serving with Istio information
+    await ops_test.model.applications["knative-serving"].set_config(
+        {
+            "istio.gateway.name": ISTIO_GATEWAY_NAME,
+            "istio.gateway.namespace": ops_test.model.name,
+        }
+    )
+    await ops_test.model.wait_for_idle(status="active", timeout=1000, idle_period=90)
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    """Deploy the autoscaling-model-server bundle."""
+    await cli_deploy_bundle(ops_test, bundle_path=BUNDLE_PATH)
+    await ops_test.model.wait_for_idle(
+        status="active", timeout=1000, idle_period=90, raise_on_error=False
+    )
+
+
+def test_inference_service_serverless_deployment(
+    ops_test: OpsTest, lightkube_client: lightkube.Client
+):
+    """Create an InferenceService and validate it has a status."""
+    # Use the model namespace for deploying the ISVC
+    serverless_namespace = ops_test.model.name
+
+    # Create InferenceService from example file
+    @tenacity.retry(
+        wait=tenacity.wait_exponential(multiplier=1, min=1, max=15),
+        stop=tenacity.stop_after_delay(30),
+        reraise=True,
+    )
+    def create_inf_svc():
+        lightkube_client.create(SKLEARN_INF_SVC_OBJECT, namespace=serverless_namespace)
```

Comment:
Couldn't this be applying multiple times the same ISVC?

Reply:
Not really, if the `create` method fails, there won't be any ISVC in the cluster.